File: JungBin/폰켓몬.cpp
#include <vector>
#include<algorithm>
using namespace std;

int solution(vector<int> nums)
{
    int answer = 0;
    sort(nums.begin(),nums.end());
    
    int index=0;
    int before=nums[0];
    answer++;
    
    while(answer!=nums.size()/2)
    {
     index++;
        
    if(index>=nums.size())
            break;
        
     if(before==nums[index])
         continue;
    
        before=nums[index];
        answer++;
    }
    
    
    return answer;
}

File: SungBin/폰켓몬.java
import java.util.*;

class Solution {
    // 15 : 20 9min
    public int solution(int[] nums) {
        int size = nums.length / 2 - 1;
        int cnt = 1;
        Arrays.sort(nums);
        
        for(int i = 0; i < nums.length-1; i++){
            if(size == 0) break;
            if(nums[i] != nums[i+1]){
                size--;
                cnt++;
            }
        }
        
        return cnt;
    }
}
